rw_mutex.go

```diff
@@ -29,8 +31,30 @@ type mongoLock struct {
 	Readers []string `bson:"readers"`
 }
 
+var emptyWriterQuery = bson.M{
+	"$or": []bson.M{
+		{
+			"writer": "",
+		},
+		{
+			"writer": nil,
+		},
+	},
+}
+
+var emptyReaderQuery = bson.M{
+	"$or": []bson.M{
+		{
+			"readers": nil,
+		},
+		{
+			"readers": []string{},
+		},
+	},
+}
+
```

Comment:
https://github.com/Clever/mongo-lock-go/pull/22/files#diff-8044f62efb4b1121122df89a0dee1859d30c8bbbcba9d7c83958b23e48eea8d8R34-R55

For this part, is this just typing what the different lock types can be?

Reply:
oh it's just what an empty read/write lock query would look like. i.e. if we needed to query mongo to see if a read lock is available, the readers portion could be nil or and empty array, so we need to check both cases.

Reply:
Gotcha thanks!